src/main/java/nl/tudelft/jpacman/npc/ghost/DispersionMove.java

```diff
@@ -0,0 +1,116 @@
+package nl.tudelft.jpacman.npc.ghost;
+
+import nl.tudelft.jpacman.board.Direction;
+import nl.tudelft.jpacman.board.Square;
+
+import java.util.List;
+
+/**
+ * The ghost dispersion move. This class implements the function <code>nextMove</code> from interface
+ * <code>{@link MoveStrategy}</code>.
+ *
+ * @author NONO TATOU P.G.
+ */
+public class DispersionMove implements MoveStrategy{
+    /**
+     * The ghost at which implement this move.
+     */
+    private Ghost ghost;
+
+    /**
+     * <code>directionCounter</code>: Useful to determine the next direction to take.
+     * <code>pathLength</code>: The length path to follow.
+     */
+    private int directionCounter, pathLength;
+
+    /**
+     * The path to follow.
+     */
+    private Direction [] cycle;
+
+    /**
+     * Creates a dispersion mode for the specified ghost.
+     *
+     * @param g
+     *      The ghost at which this will be assigned.
+     */
+    public DispersionMove(Ghost g){
+        this.ghost = g;
+        this.directionCounter = -1;
+    }
+
+    @Override
+    public Direction nextMove(){
+        Square origin = getGhost().getSquare();
+        Square home = getGhost().getHome();
+        Direction d = getGhost().randomMove();
+
+        if(origin == home || getGhost().isInMyCorner()) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539277/issues/source?bid=3256381&fileBranchId=3281825#l48)